bin/keycloak-httpd-client-install

```diff
@@ -1002,16 +1030,19 @@ def main():
                        help='certficate file')
 
     group.add_argument('--mellon-hostname', default=socket.getfqdn(),
-                       help="Machine's fully qualified host name")
+                       help="Deprecated. Use --hostname instead. ""
+                       " Machine's fully qualified host name")
 
     group.add_argument('--mellon-https-port', default=443, type=int,
```

Comment:
This comment applies to all the new common options that were derived from deprecated mellon specific options, no need to repeat this comment for each.

This isn't actually working as a common option because you're still setting and referencing the deprecated mellon variant. First take a look at the "dest" attribute for an argparse argument: [https://docs.python.org/3/library/argparse.html#dest](url). This is the option name that appears after argument parsing is complete (e.g. the argparse namespace object). So lets say the long argument is --foo, then by default you'll reference it as options.foo. But you can specify the name of the option when you call .add_argument by specifying the dest keyword. Therefore if a deprecated argument is semantically identical (e.g. mellon-hostname and hostname) you should set the dest in the deprecated argument to match the dest of it's replacement. Then only reference the new option name because both the deprecated arg and the new arg will be using the exact same parsed option name. There are numerous places where you have to fix the code for this.

Also, adding deprecated to the help is useful but only if you read the help, most people won't or have scripts that simply use the option, they'll never know the option is deprecated. Instead at run time if someone uses a deprecated option the tool need to emit a warning message which should include suggestions on how to migrate from the old to the new.

I spent a bunch of time trying to figure out the most elegant way to do this with argparse (using decorators, callbacks, adding new keyword args, `automatically` invoking the new arg, etc.) and due to architectural limitations in argparse the simplist solution I came up with is the following, edit the deprecated arg like this: 

1. add dest keyword to match new name
2. Set the action to a class that both emits a deprecation message and stores the value in dest
3. add  a new attribute to the arg object called "deprecation_help" (used by the new action class). Unfortunately you have to add the deprecation_help attribute after the arg is created, you can't add it as a keyword.

You'll need an action class for each type action, in the example snippet I'm providing I show the the standard "store" and "append" actions. Depending on what is being deprecated you might need to add another class to handle it (e.g. choice).

```python
import copy
import six

class ActionStoreDeprecated(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        deprecation_help = getattr(self, 'deprecation_help', None)
        if deprecation_help is None:
            print('arguments %s are deprecated, '
                  'no further information available' %
                  ('|'.join(self.option_strings)),
                  file=sys.stderr)
        else:
            print('arguments %s are deprecated, %s' %
                  ('|'.join(self.option_strings), deprecation_help),
                  file=sys.stderr)
        setattr(namespace, self.dest, values)


class ActionAppendDeprecated(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        deprecation_help = getattr(self, 'deprecation_help', None)
        if deprecation_help is None:
            print('arguments %s are deprecated, '
                  'no further information available' %
                  ('|'.join(self.option_strings)),
                  file=sys.stderr)
        else:
            print('arguments %s are deprecated, %s' %
                  ('|'.join(self.option_strings), deprecation_help),
                  file=sys.stderr)

        if getattr(namespace, self.dest, None) is None:
            setattr(namespace, self.dest, [])
        items = copy.copy(getattr(namespace, self.dest))
        items.append(values)
        setattr(namespace, self.dest, items)

````

So here is an example of --foo being replaced by --bar:

```python
a = parser.add_argument('--foo', '-f', dest='bar', action=ActionStoreDeprecated)
a.deprecation_help = 'use --bar|-b instead'
```